features/mbedtls/targets/TARGET_CRYPTOCELL310/ecdsa_alt.c

```diff
@@ -0,0 +1,325 @@
+        if ( signParams == NULL)
+            return ( MBEDTLS_ERR_ECP_ALLOC_FAILED );
+        pHeap = signParams;
+        heapSize = sizeof(cc_ecc_ws_sign_params_t);
+
+        pSignature = mbedtls_calloc( 1, signature_size_for_heap );
+        if ( pSignature == NULL)
+        {
+            ret = MBEDTLS_ERR_ECP_ALLOC_FAILED;
+            goto cleanup;
+        }
+
+        MBEDTLS_MPI_CHK( mbedtls_mpi_write_binary( d, temp_buf, mbedtls_mpi_size( d ) ) );
+
+        CrysRet = CRYS_ECPKI_BuildPrivKey( pDomain,
+                                       temp_buf,
+                                       mbedtls_mpi_size( d ),
+                                       &signParams->privKey);
+        if( CrysRet != CRYS_OK )
+        {
+            ret = convert_CrysError_to_mbedtls_err( CrysRet );
+            mbedtls_zeroize( temp_buf, sizeof(temp_buf) );
+            goto cleanup;
+        }
+
+        CrysRet =  CRYS_ECDSA_Sign( &cc_rand,
+                                convert_mbedtls_to_cc_rand,
+                                &signParams->signContext,
+                                &signParams->privKey,
+                                hash_mode,
+                                (uint8_t*)buf,
+                                blen,
+                                pSignature,
+                                &signature_size );
+        mbedtls_zeroize( temp_buf, sizeof(temp_buf) );
+        if( CrysRet != CRYS_OK )
+        {
+            ret = convert_CrysError_to_mbedtls_err( CrysRet );
+            goto cleanup;
+        }
+    }
+    else
+    {
+        ret =  MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE;
+        goto cleanup;
+    }
+
+    MBEDTLS_MPI_CHK( mbedtls_mpi_read_binary( r, pSignature, ( ( grp->nbits + 7 ) / 8 ) ) );
+
+    MBEDTLS_MPI_CHK( mbedtls_mpi_read_binary( s, pSignature + ( ( grp->nbits + 7 ) / 8 ), ( ( grp->nbits + 7 ) / 8 ) ) );
+
+
+cleanup:
+
+    if ( pHeap )
+    {
+        mbedtls_zeroize( pHeap, heapSize );
+        mbedtls_free( pHeap );
+    }
+
+    if( pSignature )
+    {
+        mbedtls_zeroize( pSignature, signature_size_for_heap );
+        mbedtls_free( pSignature );
+
+    }
+
+    return ( ret ) ;
+}
+#endif /* MBEDTLS_ECDSA_SIGN_ALT*/
+
+#if defined(MBEDTLS_ECDSA_VERIFY_ALT)
+//need to normalize the coordinates
+int mbedtls_ecdsa_verify( mbedtls_ecp_group *grp,
+                  const unsigned char *buf, size_t blen,
+                  const mbedtls_ecp_point *Q, const mbedtls_mpi *r, const mbedtls_mpi *s)
+{
+    int ret = 0;
+    CRYSError_t CrysRet = CRYS_OK;
+    void* pHeap = NULL;
+    size_t heapSize = 0;
+    uint8_t * pSignature = NULL;
+    CRYS_ECPKI_HASH_OpMode_t hash_mode = message_size_to_hash_mode( blen );
+    size_t temp_size = 0;
+    uint32_t signature_size = ( ( grp->nbits + 7 ) / 8 ) * 2;
+    const CRYS_ECPKI_Domain_t*  pDomain =  CRYS_ECPKI_GetEcDomain ( convert_mbedtls_grp_id_to_crys_domain_id( grp->id ) );
+
+    if( blen > 0xFFFFFFFF )
```

Comment:
Same as before.

Reply:
ok, probably a copy \ paste error. But this is not blocking, in my opinion